vulcanai2/models/basenetwork.py

```diff
@@ -291,31 +290,26 @@ def _init_optimizer(self, optim_spec):
     def _init_criterion(criterion_spec):
         return criterion_spec()
 
-    def _init_trainer(self):
+    def _init_trainer(self, train_loader):
         self.optim = self._init_optimizer(self._optim_spec)
         self.criterion = self._init_criterion(self._criter_spec)
-        self.epoch = 0
+        self.valid_interv = 2 * len(train_loader)
```

Comment:
we can remove the self.valid_interv since it is now just passed to .fit. would simplify the function signature to remove it as well